esphome/components/packet_transport/binary_sensor.py

```diff
@@ -1,19 +1,83 @@
 import esphome.codegen as cg
 from esphome.components import binary_sensor
-from esphome.const import CONF_ID
+import esphome.config_validation as cv
+from esphome.const import (
+    CONF_DATA,
+    CONF_ID,
+    CONF_NAME,
+    CONF_STATUS,
+    CONF_TYPE,
+    DEVICE_CLASS_CONNECTIVITY,
+    ENTITY_CATEGORY_DIAGNOSTIC,
+)
+import esphome.final_validate as fv
 
 from . import (
+    CONF_ENCRYPTION,
+    CONF_PING_PONG_ENABLE,
     CONF_PROVIDER,
+    CONF_PROVIDERS,
     CONF_REMOTE_ID,
     CONF_TRANSPORT_ID,
+    PacketTransport,
     packet_transport_sensor_schema,
+    provider_name_validate,
+)
+
+STATUS_SENSOR_SCHEMA = binary_sensor.binary_sensor_schema(
+    device_class=DEVICE_CLASS_CONNECTIVITY,
+    entity_category=ENTITY_CATEGORY_DIAGNOSTIC,
+).extend(
+    {
+        cv.GenerateID(CONF_TRANSPORT_ID): cv.use_id(PacketTransport),
+        cv.Required(CONF_PROVIDER): provider_name_validate,
+    }
+)
+
+CONFIG_SCHEMA = cv.typed_schema(
+    {
+        CONF_DATA: packet_transport_sensor_schema(binary_sensor.binary_sensor_schema()),
+        CONF_STATUS: STATUS_SENSOR_SCHEMA,
+    },
+    key=CONF_TYPE,
+    default_type=CONF_DATA,
 )
 
-CONFIG_SCHEMA = packet_transport_sensor_schema(binary_sensor.binary_sensor_schema())
+
+def _final_validate(config):
+    if config[CONF_TYPE] != CONF_STATUS:
+        # Only run this validation if a status sensor is being configured
+        return config
+    full_config = fv.full_config.get()
+    transport_path = full_config.get_path_for_id(config[CONF_TRANSPORT_ID])[:-1]
+    transport_config = full_config.get_config_for_path(transport_path)
+    ping_pong_enabled = transport_config.get(CONF_PING_PONG_ENABLE, False)
+    provider_encryption_enabled = True
+    if provider := next(
+        p
+        for p in transport_config.get(CONF_PROVIDERS)
+        if p.get(CONF_NAME) == config[CONF_PROVIDER]
+    ):
+        if CONF_ENCRYPTION not in provider:
+            provider_encryption_enabled = False
+    else:
+        provider_encryption_enabled = False
+    if not (ping_pong_enabled and provider_encryption_enabled):
+        raise cv.Invalid(
+            f"Status sensor {config[CONF_ID]} requires provider {config[CONF_PROVIDER]} to have enctryption defined and ping-pong enabled for {config[CONF_TRANSPORT_ID]}"
+        )
```

Comment:
`next(` requires a default argument or it throws an exception if not found, which is never acceptable when validating the yaml. When testing, you need to test the failure paths as well as the success paths. That _can_ be done in ESPHome CI unit testing, but at a minimum do it locally before pushing the changes.

Printing the ids in the error message isn't useful because if the user hasn't provided specific ids they don't mean much.

The logic is a bit convoluted and can be simplified:

```suggestion
    if transport_config[CONF_PING_PONG_ENABLE] and any(
        CONF_ENCRYPTION in p
        for p in transport_config[CONF_PROVIDERS]
        if p[CONF_NAME] == config[CONF_PROVIDER]
    ):
        return config
    raise cv.Invalid(
        "Status sensor requires ping-pong to be enabled and the nominated provider to use encryption."
    )
```